File: backend/dispatch/src/main/java/com/commonsengine/dispatch/infrastructure/persistence/DispatchRepositories.java
package com.commonsengine.dispatch.infrastructure.persistence;

import com.commonsengine.dispatch.domain.Model.TaskStatus;
import com.commonsengine.dispatch.infrastructure.persistence.DispatchPersistence.DispatchTaskEntity;
import com.commonsengine.dispatch.infrastructure.persistence.DispatchPersistence.WorkerPreferencesEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

/**
 * 调度域 JPA Repository 接口集合（匹配 Kotlin DispatchRepositories.kt）。
 */
public final class DispatchRepositories {

    private DispatchRepositories() {
    }

    public interface DispatchTaskRepository extends JpaRepository<DispatchTaskEntity, String> {
        List<DispatchTaskEntity> findByStatusOrderByCreatedAtAsc(TaskStatus status);

        List<DispatchTaskEntity> findByAssignedWorkerIdOrderByCreatedAtDesc(String workerId);
    }

    public interface WorkerPreferencesRepository extends JpaRepository<WorkerPreferencesEntity, String> {
    }
}

File: backend/dispatch/src/main/kotlin/com/commonsengine/dispatch/infrastructure/persistence/DispatchRepositories.kt
package com.commonsengine.dispatch.infrastructure.persistence

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface DispatchTaskRepository : JpaRepository<DispatchTaskEntity, Long> {

    /** 查询任务 by task_id */
    fun findByTaskId(taskId: String): DispatchTaskEntity?

    /** 查询某劳动者的所有任务 */
    fun findByWorkerId(workerId: String): List<DispatchTaskEntity>
}

@Repository
interface WorkerPreferencesRepository : JpaRepository<WorkerPreferencesEntity, Long> {

    /** 查询某劳动者的偏好（唯一） */
    fun findByWorkerId(workerId: String): WorkerPreferencesEntity?

    /** 检查是否存在 */
    fun existsByWorkerId(workerId: String): Boolean
}
